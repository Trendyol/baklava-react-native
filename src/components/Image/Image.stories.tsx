import { ComponentMeta, ComponentStory } from '@storybook/react-native';
import React from 'react';
import Box from '../Box/Box';
import Text from '../Text/Text';
import Image from './Image';

const ImageMeta: ComponentMeta<typeof Image> = {
  title: 'Image',
  component: Image,
  argTypes: {
    onPress: { action: 'pressed the button' },
  },
  args: {
    width: 48,
    height: 72,
  },
};

export default ImageMeta;

type ImageStory = ComponentStory<typeof Image>;

export const Basic: ImageStory = args => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Image
    </Text>
    <Image
      width={Number(args.width)}
      height={Number(args.height)}
      source={{
        uri: 'https://is1-ssl.mzstatic.com/image/thumb/Purple126/v4/44/e5/9c/44e59caf-c7f4-90c0-fac6-2b22d3a66a5a/AppIcon-1x_U007emarketing-0-10-0-sRGB-85-220.png/1200x600wa.png',
      }}
    />
  </>
);

export const Cases: ImageStory = () => {
  return (
    <>
      <Text p="2xs" variant="subtitle01Bold">
        Image Cases
      </Text>

      <Box px="m" py="2xs">
        <Image
          width={60}
          height={60}
          source={{
            uri: 'https://is1-ssl.mzstatic.com/image/thumb/Purple126/v4/44/e5/9c/44e59caf-c7f4-90c0-fac6-2b22d3a66a5a/AppIcon-1x_U007emarketing-0-10-0-sRGB-85-220.png/1200x600wa.png',
          }}
        />
      </Box>
      <Box px="m" py="2xs">
        <Image
          resizeMode="contain"
          source={require('../../../assets/trendyol.png')}
        />
      </Box>
      <Box px="m" py="2xs">
        <Image
          width={65}
          height={65}
          bordered={false}
          source={require('../../../assets/trendyol.png')}
        />
      </Box>
      <Box px="m" py="2xs">
        <Image source={require('../../../assets/trendyol.png')} />
      </Box>
    </>
  );
};
