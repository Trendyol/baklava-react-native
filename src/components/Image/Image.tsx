import React from 'react';
import { Image as RNImage, ImageSourcePropType } from 'react-native';
import Box from '../Box/Box';

type ImageProps = React.ComponentProps<typeof RNImage> & {
  bordered?: boolean;
  width?: number;
  height?: number;
  source: ImageSourcePropType;
  testID?: string;
  accessibilityLabel?: string;
  borderRadius?: number;
  style?: any;
};

const Image = ({
  bordered = true,
  width = 48,
  height = 72,
  source,
  testID = 'image',
  accessibilityLabel = 'image',
  borderRadius = 0,
  style,
  ...rest
}: ImageProps) => {
  return (
    <Box
      width={width}
      height={height}
      borderWidth={bordered ? 1 : 0}
      borderColor={bordered ? 'neutralLighter' : 'transparent'}
      testID={`${testID}-box`}
      style={{ borderRadius }}
      accessibilityLabel={`${accessibilityLabel}-box`}>
      <RNImage
        style={{
          height: bordered ? height - 2 : height,
          width: bordered ? width - 2 : width,
          borderRadius,
          ...style,
        }}
        source={source}
        testID={testID}
        accessibilityLabel={accessibilityLabel}
        {...rest}
      />
    </Box>
  );
};

export default Image;
