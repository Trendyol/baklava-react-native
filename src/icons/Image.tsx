import * as React from 'react';
import Svg, { SvgProps, Path } from 'react-native-svg';

const SvgImage = (props: SvgProps) => (
  <Svg
    width={24}
    height={24}
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}>
    <Path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M20.4 0H3.6C1.56 0 0 1.56 0 3.6v16.8C0 22.44 1.56 24 3.6 24h16.8c2.04 0 3.6-1.56 3.6-3.6V3.6C24 1.56 22.44 0 20.4 0Zm.6 13.68L19.32 12c-1.44-1.32-3.72-1.32-5.04 0l-1.08 1.08L9.72 9.6C8.28 8.28 6 8.28 4.68 9.6L3 11.28V4a1 1 0 0 1 1-1h16a1 1 0 0 1 1 1v9.68ZM15 7.5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0Z"
      fill={props?.fill ?? `#273142`}
    />
  </Svg>
);

export default SvgImage;
