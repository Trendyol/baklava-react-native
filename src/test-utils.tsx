import { ThemeProvider } from '@ergenekonyigit/restyle';
import { render } from '@testing-library/react-native';
import * as React from 'react';
import {
  GestureResponderHandlers,
  PanResponder,
  PanResponderCallbacks,
} from 'react-native';
import theme from './theme';
import { PortalProvider } from '.';

theme.fonts = {
  light: 'Rubik-Light',
  regular: 'Rubik-Regular',
  medium: 'Rubik-Medium',
  semiBold: 'Rubik-SemiBold',
  bold: 'Rubik-Bold',
};

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <PortalProvider>{children}</PortalProvider>
  </ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

const mockPanResponder = () => {
  jest
    .spyOn(PanResponder, 'create')
    .mockImplementation(
      ({
        onStartShouldSetPanResponderCapture,
        onMoveShouldSetPanResponder,
        onPanResponderGrant,
        onPanResponderMove,
        onPanResponderRelease,
      }: PanResponderCallbacks) => ({
        panHandlers: {
          onStartShouldSetResponderCapture: onStartShouldSetPanResponderCapture,
          onMoveShouldSetResponder: onMoveShouldSetPanResponder,
          onResponderGrant: onPanResponderGrant,
          onResponderMove: onPanResponderMove,
          onResponderRelease: onPanResponderRelease,
        } as GestureResponderHandlers,
      }),
    );
};

export * from '@testing-library/react-native';
export { customRender as render, mockPanResponder };
