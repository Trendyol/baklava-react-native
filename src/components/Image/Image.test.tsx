import React from 'react';
import { render } from '../../test-utils';
import Image from './Image';

describe('Image', () => {
  test('should render Image correctly', () => {
    // when
    const { toJSON } = render(
      <Image
        source={{
          uri: 'https://is1-ssl.mzstatic.com/image/thumb/Purple126/v4/44/e5/9c/44e59caf-c7f4-90c0-fac6-2b22d3a66a5a/AppIcon-1x_U007emarketing-0-10-0-sRGB-85-220.png/1200x600wa.png',
        }}
        accessibilityLabel="image"
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render given Image source correctly', () => {
    // when
    const { getByTestId } = render(
      <Image
        source={{
          uri: 'https://is1-ssl.mzstatic.com/image/thumb/Purple126/v4/44/e5/9c/44e59caf-c7f4-90c0-fac6-2b22d3a66a5a/AppIcon-1x_U007emarketing-0-10-0-sRGB-85-220.png/1200x600wa.png',
        }}
      />,
    );
    const imageComponent = getByTestId('image');

    // then
    expect(imageComponent.props.source).toEqual({
      uri: 'https://is1-ssl.mzstatic.com/image/thumb/Purple126/v4/44/e5/9c/44e59caf-c7f4-90c0-fac6-2b22d3a66a5a/AppIcon-1x_U007emarketing-0-10-0-sRGB-85-220.png/1200x600wa.png',
    });
  });

  test('should render Image height correctly', () => {
    // when
    const { getByTestId } = render(
      <Image
        source={{
          uri: 'https://is1-ssl.mzstatic.com/image/thumb/Purple126/v4/44/e5/9c/44e59caf-c7f4-90c0-fac6-2b22d3a66a5a/AppIcon-1x_U007emarketing-0-10-0-sRGB-85-220.png/1200x600wa.png',
        }}
        height={55}
      />,
    );
    const imageComponent = getByTestId('image');

    // then
    expect(imageComponent.props.style.height).toBe(53);
  });

  test('should render Image correctly with border', () => {
    // when
    const { getByTestId } = render(
      <Image
        source={{
          uri: 'https://is1-ssl.mzstatic.com/image/thumb/Purple126/v4/44/e5/9c/44e59caf-c7f4-90c0-fac6-2b22d3a66a5a/AppIcon-1x_U007emarketing-0-10-0-sRGB-85-220.png/1200x600wa.png',
        }}
      />,
    );
    const imageBoxComponent = getByTestId('image-box');

    // then
    expect(imageBoxComponent.props.style[0].borderWidth).toBe(1);
    expect(imageBoxComponent.props.style[0].borderColor).toBe('#AFBBCA');
  });

  test('should render Image correctly without border', () => {
    // when
    const { getByTestId } = render(
      <Image
        source={{
          uri: 'https://is1-ssl.mzstatic.com/image/thumb/Purple126/v4/44/e5/9c/44e59caf-c7f4-90c0-fac6-2b22d3a66a5a/AppIcon-1x_U007emarketing-0-10-0-sRGB-85-220.png/1200x600wa.png',
        }}
        bordered={false}
      />,
    );
    const imageBoxComponent = getByTestId('image-box');

    // then
    expect(imageBoxComponent.props.style[0].borderWidth).toBe(0);
    expect(imageBoxComponent.props.style[0].borderColor).toBe('transparent');
  });
});
