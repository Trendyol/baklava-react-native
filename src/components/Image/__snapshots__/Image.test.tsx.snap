// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render Image correctly 1`] = `
<View
  accessibilityLabel="image-box"
  accessible={true}
  style={
    Array [
      Object {
        "borderColor": "#AFBBCA",
        "borderWidth": 1,
        "height": 72,
        "width": 48,
      },
      Object {
        "borderRadius": 0,
      },
    ]
  }
  testID="image-box"
>
  <Image
    accessibilityLabel="image"
    source={
      Object {
        "uri": "https://is1-ssl.mzstatic.com/image/thumb/Purple126/v4/44/e5/9c/44e59caf-c7f4-90c0-fac6-2b22d3a66a5a/AppIcon-1x_U007emarketing-0-10-0-sRGB-85-220.png/1200x600wa.png",
      }
    }
    style={
      Object {
        "borderRadius": 0,
        "height": 70,
        "width": 46,
      }
    }
    testID="image"
  />
</View>
`;
